tests/xtro-sharpie/RequiresSuperCheck.cs

```diff
@@ -0,0 +1,66 @@
+﻿//
+// The rule reports
+//
+// !missing-requires-super!
+//		when a managed member is missing its [RequiresSuper] attribute
+//
+// !extra-designated-initializer!
+//		when a managed member has no business to have an [RequiresSuper] attribute
+//
+
+using System;
+using System.Collections.Generic;
+
+using Mono.Cecil;
+
+using Clang.Ast;
+
+namespace Extrospection {
+
+	public class RequiresSuperCheck : BaseVisitor {
+
+		static Dictionary<string,MethodDefinition> methods = new Dictionary<string,MethodDefinition> ();
+
+		static MethodDefinition GetMethod (ObjCMethodDecl decl)
+		{
+			methods.TryGetValue (decl.GetName (), out var md);
+			return md;
+		}
+
+		public override void VisitManagedMethod (MethodDefinition method)
+		{
+			var key = method.GetName ();
+			if (key == null)
```

Comment:
No idea if it is possible, but could key == ""?

Reply:
No, it's either null (if `method` is `null`) or a `StringBuilder` is created and the code makes sure it has _minimally_ (even not logical) a `::` inside it